src/Lykke.AlgoStore.CSharp.AlgoTemplate.Models/Repositories/AlgoClientInstanceRepository.cs

```diff
@@ -154,21 +156,56 @@ public async Task SaveAlgoInstanceDataAsync(AlgoClientInstanceData data)
             var clientIdPartitionKeyEntity = data.ToEntityWithClientIdPartitionKey();
             var algoIdAndClientIdPartitionKeyEntity = data.ToEntityWithAlgoIdAndClientIdPartitionKey();
             var algoIdAndInstanceTypePartitionKeyEntity = data.ToEntityWithAlgoIdAndInstanceTypePartitionKey();
-            var authTokenPartitionKeyEntity = data.ToEntityWithAuthTokenPartitionKey();        
-            AlgoInstanceStoppingEntity endDatePartitionKeyEntity;
+            var authTokenPartitionKeyEntity = data.ToEntityWithAuthTokenPartitionKey();
 
-            if (!string.IsNullOrEmpty(data.WalletId))
-            {
-                var walletIdPartitionKeyEntity = data.ToEntityWithWalletIdPartitionKey();
-                await _table.InsertOrMergeAsync(walletIdPartitionKeyEntity);
-            }
+            await SaveAlgoInstanceWalletEntity(data);
 
             await _table.InsertOrMergeAsync(algoIdPartitionKeyEntity);
             await _table.InsertOrMergeAsync(clientIdPartitionKeyEntity);
             await _table.InsertOrMergeAsync(algoIdAndClientIdPartitionKeyEntity);
             await _table.InsertOrMergeAsync(algoIdAndInstanceTypePartitionKeyEntity);
             await _table.InsertOrMergeAsync(authTokenPartitionKeyEntity);
 
+            await SaveAlgoInstanceStoppingEntity(data);
+
+            await SaveAlgoInstanceTCEntity(data);
+        }
+
+        public async Task SaveAlgoInstanceWithNewPKAsync(AlgoClientInstanceData data)
+        {
+            var clientIdPartitionKeyEntity = data.ToEntityWithClientIdPartitionKey();
+            var algoIdAndClientIdPartitionKeyEntity = data.ToEntityWithAlgoIdAndClientIdPartitionKey();
+
+            await _table.DeleteIfExistAsync(clientIdPartitionKeyEntity.PartitionKey, clientIdPartitionKeyEntity.RowKey);
+            await _table.DeleteIfExistAsync(algoIdAndClientIdPartitionKeyEntity.PartitionKey, algoIdAndClientIdPartitionKeyEntity.RowKey);
+
+            data.ClientId = _deactivatedFakeClientId;
+
+            await SaveAlgoInstanceWalletEntity(data);
+
+            var algoIdPartitionKeyEntity = data.ToEntityWithAlgoIdPartitionKey();
+            var algoIdAndInstanceTypePartitionKeyEntity = data.ToEntityWithAlgoIdAndInstanceTypePartitionKey();
+            var authTokenPartitionKeyEntity = data.ToEntityWithAuthTokenPartitionKey();
+
+            await _table.InsertOrMergeAsync(algoIdPartitionKeyEntity);
+            await _table.InsertOrMergeAsync(algoIdAndInstanceTypePartitionKeyEntity);
+            await _table.InsertOrMergeAsync(authTokenPartitionKeyEntity);
+
+            await SaveAlgoInstanceStoppingEntity(data);
+
+            await SaveAlgoInstanceTCEntity(data);
+
+            var clientIdPartitionKeyFakeEntity = data.ToEntityWithClientIdPartitionKey();
```

Comment:
No need to call the mapper twice, modify the `clientIdPartitionKeyEntity` and `algoIdAndClientIdPartitionKeyEntity` and insert them.

Reply:
they are different, I need the new one in order to have the updated Primary key, which is generated when is called ToEntityWithClientIdPartitionKey
First I delete  the old row,